internal/cloud/realm/app.go

```diff
@@ -11,10 +11,98 @@ import (
-	appPathPattern  = appsPathPattern + "/%s"
+	appsPathPattern           = adminAPI + "/groups/%s/apps"
+	appPathPattern            = appsPathPattern + "/%s"
+	appDescriptionPathPattern = appPathPattern + "/description"
 )
 
+// DataSourceSummary is a short summary for a data source model
+type DataSourceSummary struct {
+	Name       string `json:"name"`
+	Type       string `json:"type"`
+	DataSource string `json:"data_source"`
+}
+
+// HTTPEndpointSummary is a short summary for a http endpoint model
+type HTTPEndpointSummary struct {
+	Name string `json:"name"`
+}
+
+// ServiceSummary is a short summary for a service desc model
+type ServiceSummary struct {
+	Name string `json:"name"`
+	Type string `json:"type"`
+}
+
+// AuthProviderSummary is a short summary for a auth provider config model
+type AuthProviderSummary struct {
+	Name    string `json:"name"`
+	Type    string `json:"type"`
+	Enabled bool   `json:"enabled"`
+}
+
+// CustomUserDataSummary is a short summary for a custom user data config model
+type CustomUserDataSummary struct {
+	Enabled     bool   `json:"enabled"`
+	DataSource  string `json:"data_source"`
+	Database    string `json:"database"`
+	Collection  string `json:"collection"`
+	UserIDField string `json:"user_id_field"`
+}
+
+// HostingSummary is a short summary for a hosting model
+type HostingSummary struct {
+	Enabled bool   `json:"enabled"`
+	Status  string `json:"status"`
+	URL     string `json:"url"`
+}
+
+// FunctionSummary is a short summary for a function model
+type FunctionSummary struct {
+	Name string `json:"name"`
+	Path string `json:"path"`
+}
+
+// SyncSummary is a short summary for a sync model
+type SyncSummary struct {
+	State                  string `json:"state"`
+	DataSource             string `json:"data_source"`
+	Database               string `json:"database"`
+	DevelopmentModeEnabled bool   `json:"development_mode_enabled"`
+}
+
+// GraphQLSummary is a short summary for a graphql model
+type GraphQLSummary struct {
+	URL             string   `json:"url"`
+	CustomResolvers []string `json:"custom_resolvers"`
+}
+
+// EventSubscriptionSummary is a short summary for a event subscription model
+type EventSubscriptionSummary struct {
+	Name    string `json:"name"`
+	Type    string `json:"type"`
+	Enabled bool   `json:"enabled"`
+}
+
+// AppDescription describes an App
+type AppDescription struct {
+	ClientAppID       string                     `json:"client_app_id"`
+	Name              string                     `json:"name"`
+	RealmURL          string                     `json:"realm_url"`
+	DataSources       []DataSourceSummary        `json:"data_sources"`
+	HTTPEndpoints     []HTTPEndpointSummary      `json:"http_endpoints"`
+	ServiceDescs      []ServiceSummary           `json:"services"`
+	AuthProviders     []AuthProviderSummary      `json:"auth_providers"`
+	CustomUserData    CustomUserDataSummary      `json:"custom_user_data"`
+	Values            []string                   `json:"values"`
```

Comment:
[q] Are we intentionally showing only the names of values here? (i.e. not secrets or environment variables)?

Reply:
That's correct, just supposed to be kinda like a summary rather than an in-depth look is my understanding.